File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        hash_map = {}
        for x in s:
            hash_map[x] = hash_map.get(x, 0) + 1
        for x in t:
            hash_map[x] = hash_map.get(x, 0) - 1
            if hash_map[x] < 0:
                return False
        return True

File: 0242-valid-anagram/0242-valid-anagram.java

class Solution {
    public boolean isAnagram(String s, String t) {

        if(s.length()!=t.length()){
            return false;
        }

        Map<Character, Integer> occur = new HashMap<>();

        for(char x : s.toCharArray()){
            if(occur.containsKey(x)){
                occur.put(x, occur.get(x)+1);
            } else{
                occur.put(x, 1);
            }
        }

        for(char x : t.toCharArray()){
            if(!occur.containsKey(x) || occur.get(x)<=0){
                return false;
            } else{
                occur.put(x, occur.get(x)-1);
            }
        }

        return true;
        
    }
}